src/peergos/shared/io/ipfs/api/NamedStreamable.java

```diff
@@ -21,17 +24,35 @@
 
     class FileWrapper implements NamedStreamable {
         private final File source;
+        private final String pathPrefix;
 
-        public FileWrapper(File source) {
+        public FileWrapper(String pathPrefix, File source) {
```

Comment:
Suggest renaming  ``pathPrefix`` to ``remotePath`` since it isn't a prefix, Also if you're doing file-path manipulations (get-parent  etc.)  perhaps use a ``java.nio.file.Path`` instead of a  string?